src/main/java/net/minestom/server/entity/metadata/display/AbstractDisplayMeta.java

```diff
@@ -0,0 +1,141 @@
+package net.minestom.server.entity.metadata.display;
+
+import net.minestom.server.coordinate.Point;
+import net.minestom.server.coordinate.Vec;
+import net.minestom.server.entity.Entity;
+import net.minestom.server.entity.Metadata;
+import net.minestom.server.entity.metadata.EntityMeta;
+import net.minestom.server.utils.Quaternion;
+import org.jetbrains.annotations.NotNull;
+
+public class AbstractDisplayMeta extends EntityMeta {
+
+    public static final byte OFFSET = EntityMeta.MAX_OFFSET;
+    public static final byte MAX_OFFSET = OFFSET + 14;
+
+    protected AbstractDisplayMeta(@NotNull Entity entity, @NotNull Metadata metadata) {
+        super(entity, metadata);
+    }
+
+    public long getInterpolationStartTicks() {
+        return super.metadata.getIndex(OFFSET, -1000L);
+    }
+
+    public void setInterpolationStartTicks(long value) {
+        super.metadata.setIndex(OFFSET, Metadata.VarLong(value));
+    }
+
+    public int getInterpolationDuration() {
+        return super.metadata.getIndex(OFFSET + 1, 0);
+    }
+
+    public void setInterpolationDuration(int value) {
+        super.metadata.setIndex(OFFSET + 1, Metadata.VarInt(value));
+    }
+
+    public @NotNull Point getTranslation() {
+        return super.metadata.getIndex(OFFSET + 2, Vec.ZERO);
+    }
+
+    public void setTranslation(@NotNull Point value) {
+        super.metadata.setIndex(OFFSET + 2, Metadata.Vector3(value));
+    }
+
+    public @NotNull Vec getScale() {
+        return super.metadata.getIndex(OFFSET + 3, Vec.ONE);
+    }
```

Comment:
since Metadata.Vector3 specifies Point, casting would fail if the entry is a Pos
(I'm not really sure if this is an issue)